frontend/public/js/editperformance.js

```diff
@@ -0,0 +1,292 @@
+var selectedPerformance, nameEditingPerformance;
+
+/**
+ * Loads the edit performance window.
+ */
+function loadEditPerformance() {
+    drawPresets(presets);
+    drawPerformancesList();
+    drawSchedule();
+}
+
+/**
+ * Draws the performance list from which the performance to edit can be selected.
+ */
+function drawPerformancesList() {
+    var performancelist = $('#performance-list');
+    performancelist.empty();
+
+    for (var i in performances) {
+        var p = performances[i];
+
+        var li = drawPerformanceListItem(p);
+        performancelist.append(li);
+    }
+}
+
+/**
+ * Draws one list item from the performance list.
+ * @param performance   The performance object to draw.
+ * @returns {*|jQuery|HTMLElement} a jQuery DOM element to append.
+ */
+function drawPerformanceListItem(performance) {
+    var li = $("<li class='btn btn-info'></li>");
+    li.data(performance);
+    li.click(selectPerformance);
+    var icon = $("<span class='glyphicon glyphicon-bullhorn'> " + performance['name'] + "</span>");
+
+    li.append(icon);
+    return li;
+}
+
+/**
+ * Selects a performance
+ */
+function selectPerformance() {
+    if (nameEditingPerformance !== undefined) {
+        nameEditingPerformance.replaceWith(drawPerformanceListItem($(nameEditingPerformance).data()));
+    }
+
+    if (selectedPerformance !== undefined) {
+        selectedPerformance.attr('class', 'btn btn-info');
+    }
+
+    selectedPerformance = $(this);
+    selectedPerformance.attr('class', 'btn btn-info disabled');
+
+    drawSchedule(selectedPerformance.data());
+}
+
+function setNameEditable() {
+    if (selectedPerformance !== undefined) {
+        var element = selectedPerformance;
+
+        var li = $('<li><div class="row">' +
+            '<div class="col-xs-7"><input type="text" id="performance-name" class="form-control" value="' + $(element).data()['name'] + '"></div> ' +
```

Comment:
Since ejs is rendered in nodeJS and not on the client this is not possible.
However I agree with you that in the future the code can be refactored to be come more clean.